src/Dropdown/index.jsx

```diff
@@ -1,225 +1,239 @@
+            toggle: this.toggle,
+            triggerId: this.triggerId,
+            buttonRef: this.buttonRef,
+            ariaExpanded: this.state.open,
+            ariaHasPopup: true,
+            className: classNames({
+                show: this.state.open,
+              }),
+            handleMenuKeyDown: this.handleMenuKeyDown,
+            menuRef: this.menuRef,
+            ariaLabelledBy: this.triggerId,
+            ariaHidden: !this.state.open,
+              }}
         >
-          {this.generateMenuItems(menuItems)}
-        </div>
+          {children}
+        </Provider>
       </div>
     );
   }
 }
 
 Dropdown.propTypes = {
-  buttonType: PropTypes.string,
-  iconElement: PropTypes.element,
-  menuItems: PropTypes.arrayOf(
-    PropTypes.shape({
-      label: PropTypes.string,
-      href: PropTypes.string,
-    }),
-    PropTypes.element,
-  ).isRequired,
-  title: PropTypes.string.isRequired,
+  className: PropTypes.string,
+  buttonClassName: PropTypes.string,
+  children: PropTypes.node.isRequired,
+  buttonContent: PropTypes.node.isRequired,
 };
 
 Dropdown.defaultProps = {
-  buttonType: 'light',
-  iconElement: undefined,
+  className: null,
+  buttonClassName: 'btn-light',
 };
 
-export default Dropdown;
+const DropdownWithDeprecatedProps = withDeprecatedProps(Dropdown, 'Dropdown', {
+  menuItems: {
+    deprType: DEPR_TYPES.MOVED_AND_FORMAT,
+    message: 'They should be components sent as children.',
+    newName: 'children',
+    transform: (menuItems, allProps) => {
+      if (!Array.isArray(menuItems)) return null;
+      return (
+        <React.Fragment>
+          <DropdownButton>
+            {React.isValidElement(allProps.iconElement) ? allProps.iconElement : null }
+            {allProps.title}
+          </DropdownButton>
+          <DropdownMenu>
+            {menuItems.map((menuItem, i) => {
+            /* eslint-disable react/no-array-index-key */
+            if (React.isValidElement(menuItem)) {
+              return React.cloneElement(menuItem, {
+                className: 'dropdown-item',
+                key: i,
+              });
+            }
+            return <DropdownItem key={i} href={menuItem.href}>{menuItem.label}</DropdownItem>;
+            /* eslint-enable react/no-array-index-key */
+            })}
+          </DropdownMenu>
+        </React.Fragment>);
+    },
+  },
+  title: {
+    deprType: DEPR_TYPES.REMOVED,
+    message: 'It should be specified inside the Dropdown.Button component',
+  },
+  buttonType: {
+    deprType: DEPR_TYPES.REMOVED,
+    message: 'It should be specified as a className prop',
+  },
+  iconElement: {
+    deprType: DEPR_TYPES.REMOVED,
+    message: 'It should be specified inside the buttonContent prop.',
+  },
+});
+
+DropdownWithDeprecatedProps.propTypes = Dropdown.propTypes;
+DropdownWithDeprecatedProps.defaultProps = Dropdown.defaultProps;
+DropdownWithDeprecatedProps.Item = DropdownItem;
+DropdownWithDeprecatedProps.Button = DropdownButton;
```

Comment:
Let's also do these assignments for Dropdown (e.g. `Dropdown.Button = DropdownButton`) so that we don't need to remember to add it in when we remove this withDeprecatedProps shim.